packages/client/src/publish/GroupKeyQueue.ts

```diff
@@ -38,16 +42,28 @@ export class GroupKeyQueue {
         return result
     }
 
-    async rotate(newKey = GroupKey.generate()): Promise<GroupKey> {
+    async rotate(newKey?: GroupKey): Promise<GroupKey> {
+        if (newKey === undefined) {
+            newKey = await this.generateNewKey()
+        }
         this.queuedGroupKey = newKey
         await this.store.add(newKey, this.streamId)
         return newKey
     }
 
-    async rekey(newKey = GroupKey.generate()): Promise<GroupKey> {
+    async rekey(newKey?: GroupKey): Promise<GroupKey> {
+        if (newKey === undefined) {
+            newKey = await this.generateNewKey()
+        }
         await this.store.add(newKey, this.streamId)
         this.currentGroupKey = newKey
         this.queuedGroupKey = undefined
         return newKey
     }
+
+    private async generateNewKey(): Promise<GroupKey> {
+        // 1st try lit-protocol, if a key cannot be generated and stored, then generate group key locally
+        const litProtocolGroupKey = await this.litProtocolKeyStore.store(this.streamId, crypto.randomBytes(32))
+        return litProtocolGroupKey !== undefined ? litProtocolGroupKey : GroupKey.generate()
```

Comment:
Something like this?
```suggestion
        const keyData =  crypto.randomBytes(32)
        // 1st try lit-protocol, if a key cannot be generated and stored, then generate group key locally
        const litProtocolGroupKey = await this.litProtocolKeyStore.store(this.streamId, keyData)
        return litProtocolGroupKey !== undefined ? litProtocolGroupKey : GroupKey.generate(keyData)
```